### PR TITLE
Fix parallel flow_cp for the interleaved case

### PR DESCRIPTION
**Pull request by blattms**
With this PR and zoltan activated via OPM/dune-cornerpoint#163 and #459 it is finally possible to at least run norne in parallel with the interleaved ordering/blocks.

### CODE REVIEW

**Comment by blattms** [comment]
BTW this is not enough if solve_welleq_initially ist set to true. In that case we will still experience problems (read segmentation faults, mpi errors...).

File: opm/autodiff/NewtonIterationBlackoilInterleaved.cpp [hunk 1]
```diff
@@ -153,8 +153,10 @@ namespace Opm
             const ParallelISTLInformation& info =
                 boost::any_cast<const ParallelISTLInformation&>( parallelInformation_);
             Comm istlComm(info.communicator());
+            // As we use a dune-istl with block size np the number of components
+            // per parallel is only one.
```
Comment (atgeirr) [inline opm/autodiff/NewtonIterationBlackoilInterleaved.cpp]: I did not quite understand the comment, which probably means I do not understand what the argument to copyValuesTo() is supposed to be?

Comment (blattms) [inline opm/autodiff/NewtonIterationBlackoilInterleaved.cpp]: Let's give it another try:

The parallelization of dune-istl exspects one pair of global (unique
over all processes) and local (0, ..., number of indices on process)
index per block. For the interleaved version this block represents all
phases. For the non-interleaved one it represents just one phase.

The indexset passed to copy values contains one index pair per
cell. This is sufficient for the interleaved solver with one block per
cell.

But for the other one we need to communicates number of phases (which
I call number_of_components here) values which is not possible for
that index set. Therefore we create a new one that has
number_of_components times the entries and use that one for the
communication.
-- 
Join us at the DUNE User Meeting Sept. 28-29, 2015, Heidelberg, Germany
http://users.dune-project.org/projects/dune-user-meeting-2015/wiki

**Comment by joakim-hove** [comment]
> With this PR and zoltan activated via OPM/dune-cornerpoint#163 and #459 it is finally possible to at least run norne in parallel with the interleaved ordering/blocks.

"Minor"(?) problems with wells notwithstanding - this is worthy of a celebration: :fireworks: :cake:

File: opm/autodiff/NewtonIterationBlackoilInterleaved.cpp (hunk 1 above)
Comment (atgeirr) [inline opm/autodiff/NewtonIterationBlackoilInterleaved.cpp]: Thanks, think I got it now. Block-structuring rocks.

**Comment by bska** [comment]
Is it too much to ask that the PR be build tested by someone other than the author prior to merging?  We're missing the header `opm/autodiff/ParallelRestrictAdditiveSchwarz.hh` (which should be named `.hpp` for consistency with the rest of the module) and `CMakeLists_files.txt` does not account for the new files meaning `make install` fails.

**Comment by alfbr** [comment]
I believe we have a PR builder on Jenkins that is about high time we put to good use, it will quickly bit-rot if not used. It is found here:
http://80.240.129.131:8080/job/OPM_multi-PR-build/

**Comment by blattms** [comment]
Sorry @bska and everyone. That was my fault. The include should never
have been committed. Please merge #462.

**Pull request merged**